File: java/revenj-servlet/src/main/java/org/revenj/server/servlet/PassThroughSerialization.java
package org.revenj.server.servlet;

import org.revenj.serialization.Serialization;

import java.io.IOException;
import java.lang.reflect.Type;

final class PassThroughSerialization implements Serialization<Object> {

	@Override
	public Object serialize(Type type, Object value) {
		return value;
	}

	@Override
	public Object deserialize(Type type, Object data) throws IOException {
		return data;
	}
}

File: scala/revenj-akka/src/main/scala/net/revenj/server/PassThroughSerialization.scala
package net.revenj.server

import java.lang.reflect.Type

import net.revenj.serialization.Serialization

import scala.util.{Success, Try}
import scala.reflect.runtime.universe.TypeTag

class PassThroughSerialization extends Serialization[Any] {
  override def serialize[T: TypeTag](value: T): Try[Any] = Success(value)

  override private[revenj] def serializeRuntime(value: Any, manifest: Type): Try[Any] = Success(value)

  override def deserialize[T: TypeTag](input: Any): Try[T] = Success(input.asInstanceOf[T])

  override private[revenj] def deserializeRuntime[T](input: Any, manifest: Type): Try[T] = Success(input.asInstanceOf[T])
}
